dora/core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java

```diff
@@ -784,9 +787,10 @@ private boolean restore(PageStoreDir pageStoreDir) {
       return false;
     }
     try {
-      pageStoreDir.scanPages(pageInfo -> {
-        if (pageInfo.isPresent()) {
-          addPageToDir(pageStoreDir, pageInfo.get());
+      pageStoreDir.scanPages(optionalPageInfo -> {
+        if (optionalPageInfo.isPresent()) {
+          PageInfo pageInfo = optionalPageInfo.get();
+          addPageBasedOnPredicate(pageStoreDir, pageInfo);
```

Comment:
do you feel it is cleaner here if we go the following way? b/c we can keep original code path unchanged, while it is still easy to follow and make `addPageBasedOnPredicate` simpler

```java

if (mPagePredicate.isPresent()) {
  addPageBasedOnPredicate(pageStoreDir, pageInfo);
} else {
  addPageToDir(pageStoreDir, pageInfo);
}

```

Reply:
done